### PR TITLE
Restructured to support externalization

### LINKED ISSUE #76: restructure new-repo and new-proj
As a developer, I want each file to describe what it does.
As a developer, I want each file to have its own responsibility.
As a developer, I want to optimize functionality.

### PR DESCRIPTION
**Pull request by veikkahonkanen**
Resolves #76

### CODE REVIEW

File: src/commands/new-proj.ts [hunk 1]
```diff
@@ -1,14 +1,22 @@
...
 
 /**
  * Prompts the user and runs corresponding commands
  */
 async function action() {
   let projName : string = null;
+  let projType : string = null;
+  let gh = false;
 
   try {
```
Comment (eetupur) [inline src/commands/new-proj.ts:21]: prompts could be in their own try-catch

File: src/commands/new-proj.ts [hunk 2]
```diff
@@ -20,20 +28,86 @@ async function action() {
...
+    //To do: Add React
+    this.log("Creating react project - please wait...");
+  }
+
+  if (projType == "No framework") {
+    this.log("Creating project - please wait...");
+    const cmds : string[] = await CreateDefault(projName, folderPath, repoPath);
+    execSync(cmds[0]);
+    execSync(cmds[1], {cwd : `.${path.sep}resources`});
+    execSync(cmds[2], {cwd : `.${path.sep}resources`});
```
Comment (eetupur) [inline src/commands/new-proj.ts]: surround with try catch

File: src/commands/new-proj.ts [hunk 3]
```diff
@@ -1,14 +1,22 @@
 import Vorpal from "vorpal";
-import { newRepo } from "./new-repo";
+import { newRepo } from "./new-proj/new-repo";
```
Comment (eetupur) [inline src/commands/new-proj.ts]: new-repo does not necessarily need to be inside new-proj folder, it is a convenient way for creating a new repo, so the user might want to use it separately from new-proj

File: src/commands/new-proj.ts [hunk 4]
```diff
@@ -20,20 +28,86 @@ async function action() {
...
+
+    const typeResult = await this.prompt({
+      type : 'list',
+      name : 'type',
+      choices : ["Quarkus", "React", "No framework"],
+      message : "Framework for the project: "
+    });
+
+    if (typeResult.name) {
+      projType= typeResult.name;
```
Comment (eetupur) [inline src/commands/new-proj.ts]:
```suggestion
    if (typeResult.type) {
      projType= typeResult.type;
```

File: src/commands/new-proj.ts [hunk 5]
```diff
@@ -20,20 +28,86 @@ async function action() {
...
+
+    if (pathResult?.path) {
+      givenPath = pathResult.path;
     }
   } catch (err) {
-    throw new Error(err);
+    throw new Error("Error: " + err);
+  }
+
+  givenPath = await PathUtils.fixPath(givenPath);
```
Comment (eetupur) [inline src/commands/new-proj.ts]: awaits should preferrably have try-catch

File: src/commands/new-proj.ts [hunk 6]
```diff
@@ -20,20 +28,86 @@ async function action() {
...
-    .use(newRepo)
-    .execSync("new-repo");
-  } catch(err) {
-    throw new Error("Encountered an error while creating repository: " + err);
+    const repoResult = await this.prompt({
+      type : 'list',
+      name : 'repo',
+      choices : ["Yes", "No"],
+      message : "Create GitHub repository for the project?: "
+    });
```
Comment (eetupur) [inline src/commands/new-proj.ts]: a gh repo is necessary for all types of project, only what settings it has and when it is created can change

File: src/commands/new-proj.ts [hunk 7]
```diff
@@ -20,20 +28,86 @@ async function action() {
...
+
+    if (repoResult.name) {
+      if (repoResult.name == "Yes") {
```
Comment (eetupur) [inline src/commands/new-proj.ts]: probably not a needed change as the prompt might be unneeded but
```suggestion
    if (repoResult.repo) {
      if (repoResult.repo == "Yes") {
```

File: src/commands/new-proj/create-default.ts [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+ * @param projName desired project name
+ * 
+ * @param folderPath outer folder
+ * 
+ * @param repoPath inner folder, for GitHub etc
+ * 
+ * @returns an array of strings which includes the commands to be run on the new-proj thread
+ */
+export const CreateDefault = async (projName : string, folderPath : string, repoPath : string) : Promise<string[]> => {
+  const copy = await OsUtils.getCommand(CommandNames.copy);
```
Comment (eetupur) [inline src/commands/new-proj/create-default.ts]: try catch for await

File: src/commands/new-proj/new-repo.ts [hunk 9]
```diff
@@ -24,12 +19,10 @@ async function action(args) {
...
     args.options.publicity :
     "private";
-
-  let givenPath : string =  args.options.path ?
-    args.options.path :
-    defaultPath;
-
-  if (!publicity || !repoName) {
+  
+  let givenPath : string =  args.options.path;
```
Comment (eetupur) [inline src/commands/new-proj/new-repo.ts]: program should use default path if none is given

**Review by eetupur — CHANGES_REQUESTED** [review]
found some issues

File: src/commands/new-proj.ts [hunk 10]
```diff
@@ -20,20 +27,71 @@ async function action() {
...
+
+    const typeResult = await this.prompt({
+      type : 'list',
+      name : 'type',
+      choices : ["Quarkus", "React", "No framework"],
+      message : "Framework for the project: "
+    });
+
+    if (typeResult.type) {
+      projType= typeResult.type;
```
Comment (eetupur) [inline src/commands/new-proj.ts]:
```suggestion
      projType = typeResult.type;
```

File: src/commands/new-proj.ts [hunk 11]
```diff
@@ -20,20 +27,71 @@ async function action() {
...
+
+  if (projType == "No framework") {
+    this.log("Creating project - please wait...");
+    try {
+      const cmds : string[] = await CreateDefault(projName, folderPath, repoPath);
+      execSync(cmds[0]);
+      execSync(cmds[1], {cwd : `.${path.sep}resources`});
+      execSync(cmds[2], {cwd : `.${path.sep}resources`});
+    } catch(err) {
+      throw new Error(err);
```
Comment (eetupur) [inline src/commands/new-proj.ts]: could have a more descriptive error

File: src/commands/new-proj.ts [hunk 12]
```diff
@@ -20,20 +27,71 @@ async function action() {
...
+      name : 'path',
+      message : "Set a path where to initiate repository, leave empty for default: "
+    });
+
+    if (pathResult?.path) {
+      givenPath = pathResult.path;
     }
   } catch (err) {
-    throw new Error(err);
+    throw new Error("Error: " + err);
```
Comment (eetupur) [inline src/commands/new-proj.ts]: could have more descriptive error

File: src/commands/new-proj/new-repo.ts [hunk 13]
```diff
@@ -20,15 +15,15 @@ async function action(args) {
   let repoName : string = args.name;
   let description : string = args.options.description;
   const template = args.options.template;
+  const { HOME } = process.env;
+  const defaultPath = `${HOME}/.meta-proj-cli/projects`;
```
Comment (eetupur) [inline src/commands/new-proj/new-repo.ts]: shouldnt we just make this a feature in path-utils at this point

**Review by eetupur — CHANGES_REQUESTED** [review]
found things to improve

File: src/commands/new-proj.ts (hunk 3 above)
Comment (veikkahonkanen) [reply src/commands/new-proj.ts]: refactored new-repo back under commands/

File: src/commands/new-proj.ts (hunk 2 above)
Comment (veikkahonkanen) [reply src/commands/new-proj.ts]: now try-catch added

File: src/commands/new-proj.ts (hunk 5 above)
Comment (veikkahonkanen) [reply src/commands/new-proj.ts]: they now have try-catch

File: src/commands/new-proj.ts (hunk 7 above)
Comment (veikkahonkanen) [reply src/commands/new-proj.ts]: prompt ended up being unnecessary

File: src/commands/new-proj/create-default.ts (hunk 8 above)
Comment (veikkahonkanen) [reply src/commands/new-proj/create-default.ts]: added try-catch

File: src/commands/new-proj/new-repo.ts (hunk 9 above)
Comment (veikkahonkanen) [reply src/commands/new-proj/new-repo.ts]: now conditionally uses default path

File: src/commands/new-proj.ts (hunk 6 above)
Comment (veikkahonkanen) [reply src/commands/new-proj.ts]: enforced usage of new-repo

File: src/commands/new-proj.ts (hunk 11 above)
Comment (veikkahonkanen) [reply src/commands/new-proj.ts]: added descriptive errors

File: src/commands/new-proj.ts (hunk 12 above)
Comment (veikkahonkanen) [reply src/commands/new-proj.ts]: added descriptive error

File: src/commands/new-proj/new-repo.ts (hunk 13 above)
Comment (veikkahonkanen) [reply src/commands/new-proj/new-repo.ts]: that is left for another issue

**Review by eetupur — APPROVED** [review]

**Review by eetupur — APPROVED** [review]

File: src/commands/new-proj.ts [hunk 14]
```diff
@@ -1,14 +1,25 @@
...
+import { runExecSync } from "../classes/exec-sync-utils";
 
 const vorpal = new Vorpal();
+const { HOME } = process.env;
+let givenPath = `${HOME}/.meta-proj-cli/projects`;
+let projName : string = null;
+let projType : string = null;
+let projVm : string = null;
+let folderPath : string = null;
+let repoPath : string = null;
```
Comment (jarinykanen) [inline src/commands/new-proj.ts]:
```suggestion
let projName: string = null;
let projType: string = null;
let projVm: string = null;
let folderPath: string = null;
let repoPath: string = null;
```

File: src/commands/new-proj.ts [hunk 15]
```diff
@@ -20,20 +31,79 @@ async function action() {
     if (nameResult.name) {
       projName= nameResult.name;
```
Comment (jarinykanen) [inline src/commands/new-proj.ts]:
```suggestion
      projName = nameResult.name;
```

File: src/commands/new-proj.ts [hunk 16]
```diff
@@ -20,20 +31,79 @@ async function action() {
...
     } else {
-      throw new Error("ERROR: No name was given for the project");
+      throw new Error("No name was given for the project");
+    }
+
+    const typeResult = await this.prompt({
+      type : 'list',
+      name : 'type',
+      choices : ["Quarkus", "React", "No framework"],
+      message : "Framework for the project: "
```
Comment (jarinykanen) [inline src/commands/new-proj.ts]:
```suggestion
      type: 'list',
      name: 'type',
      choices: ["Quarkus", "React", "No framework"],
      message: "Framework for the project: "
```

Comment (jarinykanen) [reply src/commands/new-proj.ts]: Fix this everywhere

File: src/commands/new-proj.ts [hunk 17]
```diff
@@ -47,7 +117,41 @@ async function action() {
       .execSync("test");
     }
   } catch(err) {
-    throw new Error(err);
+    throw new Error(`Error while performing tests: ${err}`);
+  }
+}
+
+async function resolvePaths() {
```
Comment (jarinykanen) [inline src/commands/new-proj.ts:127]: Add TSDocs

File: src/commands/new-proj.ts [hunk 18]
```diff
@@ -47,7 +117,41 @@ async function action() {
...
+    givenPath = await PathUtils.fixPath(givenPath);
+    folderPath = PathUtils.outerFolder(givenPath, projName);
+    repoPath = PathUtils.repoFolder(givenPath, projName);
+
+  } catch (err) {
+    throw new Error(`Error when attempting to resolve paths: ${err}`);
+  }
+}
+
+async function initDefaultProject() {
```
Comment (jarinykanen) [inline src/commands/new-proj.ts:141]: Add TSDocs

File: src/commands/new-proj.ts [hunk 19]
```diff
@@ -47,7 +117,41 @@ async function action() {
...
+    await runExecSync(cmds[0]);
+    await runExecSync(cmds[1], {cwd : `.${path.sep}resources`});
+    await runExecSync(cmds[2], {cwd : `.${path.sep}resources`});
+
+  } catch(err) {
+    throw new Error(`Error when creating project : ${err}`);
+  }
+}
+
+async function repoViaVorpal() {
```
Comment (jarinykanen) [inline src/commands/new-proj.ts:156]: Add TSDocs

File: src/commands/new-proj/create-default.ts [hunk 20]
```diff
@@ -0,0 +1,29 @@
...
+ * 
+ * @param projName desired project name
+ * 
+ * @param folderPath outer folder
+ * 
+ * @param repoPath inner folder, for GitHub etc
+ * 
+ * @returns an array of strings which includes the commands to be run on the new-proj thread
+ */
+export const CreateDefault = async (projName : string, folderPath : string, repoPath : string) : Promise<string[]> => {
```
Comment (jarinykanen) [inline src/commands/new-proj/create-default.ts]:
```suggestion
/**
 * Constructs creation command for default projects
 *
 * @param projName desired project name
 * @param folderPath outer folder
 * @param repoPath inner folder, for GitHub etc
 * @returns an array of strings which includes the commands to be run on the new-proj thread
 */
export const CreateDefault = async (projName: string, folderPath: string, repoPath: string): Promise<string[]> => {
```

File: src/commands/new-repo.ts [hunk 21]
```diff
@@ -58,66 +59,55 @@ async function action(args) {
...
-  await runExecSync(
-    `gh repo create\
-    ${process.env.GIT_ORGANIZATION}/${repoName}\
-    --${publicity}\
-    ${description ? `-d="${description}"` : ""}\
-    ${template ? `--template="${process.env.GIT_ORGANIZATION}/${template}"` : ""}\
-    -y`,
-    {cwd : folderPath}
-  );
+async function finishRepo() {
```
Comment (jarinykanen) [inline src/commands/new-repo.ts:80]: Add TSDocs

File: src/commands/new-repo.ts [hunk 22]
```diff
@@ -58,66 +59,55 @@ async function action(args) {
...
-    } catch (err) {
-      throw new Error(`Encountered error while creating local repository: ${err}`);
     }
+  } catch (err) {
+    throw new Error(`Error when initing a repository: ${err}`);
   }
 }
 
-async function finishRepo (repoPath) {
+async function checkout() {
```
Comment (jarinykanen) [inline src/commands/new-repo.ts:112]: Add TSDocs

File: test/new-repo.spec.ts [hunk 23]
```diff
@@ -44,7 +44,7 @@ it('creates project folder', () => {
   dotenv.config();
   const { HOME } = process.env;
   const defaultPath = `${HOME}/.meta-proj-cli/projects`;
-
+
```
Comment (jarinykanen) [inline test/new-repo.spec.ts]:
```suggestion
```

**Review by jarinykanen — CHANGES_REQUESTED** [review]

File: src/commands/new-proj.ts (hunk 16 above)
Comment (veikkahonkanen) [reply src/commands/new-proj.ts]: Removed ~250 of these in ~30 files

File: src/commands/new-proj.ts (hunk 17 above)
Comment (veikkahonkanen) [reply src/commands/new-proj.ts:127]: Added tsdocs

File: src/commands/new-proj.ts (hunk 18 above)
Comment (veikkahonkanen) [reply src/commands/new-proj.ts:141]: Added tsdocs

File: src/commands/new-proj.ts (hunk 19 above)
Comment (veikkahonkanen) [reply src/commands/new-proj.ts:156]: Added tsdocs

File: src/commands/new-repo.ts (hunk 21 above)
Comment (veikkahonkanen) [reply src/commands/new-repo.ts:80]: Added tsdocs

File: src/commands/new-repo.ts (hunk 22 above)
Comment (veikkahonkanen) [reply src/commands/new-repo.ts:112]: Added tsdocs

File: src/commands/add-readme.ts [hunk 24]
```diff
@@ -15,21 +15,21 @@ async function action(args: { options: { path: string; }; }) {
 
   if (args.options.path) {
     path = argPath.startsWith("~") ? 
-      HOME + argPath.slice(1) :
-      argPath;
+      HOME + argPath.slice(1)
+      : argPath;
```
Comment (eetupur) [inline src/commands/add-readme.ts]: unless you were asked to change this, i would keep the : on the previous row, as it makes it clear that this piece of code continues on the next line
```suggestion
      HOME + argPath.slice(1) :
      argPath;
```

**Review by eetupur — COMMENTED** [review]
found something questionable

File: src/classes/path-utils.ts [hunk 25]
```diff
@@ -11,36 +11,36 @@ const defaultProjectPath = "~/.meta-proj-cli/projects";
 /**
  * Offers functions that help with paths
  * 
- * fixPath : changes file separators to platform specific ones and expands tilde(~) paths
- * outerFolder : gives the path format to the outer folder of a project
- * repoFolder : gives the path format to the folder that actually holds the project (contains .git)
+ * fixPath: changes file separators to platform specific ones and expands tilde(~) paths
+ * outerFolder: gives the path format to the outer folder of a project
+ * repoFolder: gives the path format to the folder that actually holds the project (contains .git)
```
Comment (jarinykanen) [inline src/classes/path-utils.ts]: These descriptions should be on function docs not in class docs

File: src/commands/add-readme.ts [hunk 26]
```diff
@@ -19,17 +19,17 @@ async function action(args: { options: { path: string; }; }) {
...
-  await runExecSync("git init", {cwd : path});
-  await runExecSync("git add README.md", {cwd : path});
-  await runExecSync('git commit -m "first commit"', {cwd : path});
-  await runExecSync(`git remote add origin ${process.env.GIT_REPO_BASE_PATH}`, {cwd : path});
-  await runExecSync("git push -u origin master", {cwd : path});
+  await runExecSync("git init", {cwd: path});
+  await runExecSync("git add README.md", {cwd: path});
+  await runExecSync('git commit -m "first commit"', {cwd: path});
+  await runExecSync(`git remote add origin ${process.env.GIT_REPO_BASE_PATH}`, {cwd: path});
+  await runExecSync("git push -u origin master", {cwd: path});
```
Comment (jarinykanen) [inline src/commands/add-readme.ts]:
```suggestion
  await runExecSync("git init", { cwd: path });
  await runExecSync("git add README.md", { cwd: path });
  await runExecSync('git commit -m "first commit"', { cwd: path });
  await runExecSync(`git remote add origin ${process.env.GIT_REPO_BASE_PATH}`, { cwd: path} );
  await runExecSync("git push -u origin master", { cwd: path });
```

File: src/commands/add-readme.ts [hunk 27]
```diff
@@ -19,17 +19,17 @@ async function action(args: { options: { path: string; }; }) {
...
 
   if (!args.options.path) {
-    await runExecSync("rm -r .git", {cwd : path});
+    await runExecSync("rm -r .git", {cwd: path});
```
Comment (jarinykanen) [inline src/commands/add-readme.ts]:
```suggestion
    await runExecSync("rm -r .git", { cwd: path });
```

File: src/commands/checkTest.ts [hunk 28]
```diff
@@ -6,7 +6,7 @@ import { CheckSet } from "../interfaces/types";
  * Shows a debug message
  */
 async function action() {
-  const test : CheckSet[] = [{checkable : "git", details : {}}]
+  const test: CheckSet[] = [{checkable: "git", details: {}}]
```
Comment (jarinykanen) [inline src/commands/checkTest.ts]:
```suggestion
  const test: CheckSet[] = [{ checkable: "git", details: {} }]
```

File: src/classes/os-utils.ts [hunk 29]
```diff
@@ -3,7 +3,7 @@ import { OsCommands } from "./os-commands";
 import { UserConfigUtils } from "./user-config-utils";
 import { windows, linux, macos } from "platform-detect";
 
-const systems : string[]  = [OperatingSystems.MAC, OperatingSystems.WINDOWS, OperatingSystems.LINUX];
+const systems: string[]  = [OperatingSystems.MAC, OperatingSystems.WINDOWS, OperatingSystems.LINUX];
```
Comment (jarinykanen) [inline src/classes/os-utils.ts]:
```suggestion
const systems: string[]  = [ OperatingSystems.MAC, OperatingSystems.WINDOWS, OperatingSystems.LINUX ];
```

File: src/commands/new-proj.ts [hunk 30]
```diff
@@ -1,44 +1,114 @@
...
+      projName = nameResult.name;
     } else {
-      throw new Error("ERROR: No name was given for the project");
+      throw new Error("No name was given for the project");
+    }
+
+    const typeResult = await this.prompt({
+      type: 'list',
+      name: 'type',
+      choices: ["Quarkus", "React", "No framework"],
```
Comment (jarinykanen) [inline src/commands/new-proj.ts]:
```suggestion
      type: "list",
      name: "type",
      choices: [ "Quarkus", "React", "No framework" ],
```

File: src/commands/new-proj.ts [hunk 31]
```diff
@@ -1,44 +1,114 @@
...
+    if (typeResult.type) {
+      projType = typeResult.type;
+    } else {
+      throw new Error("No type was given for the project");
+    }
+
+    const vmResult = await this.prompt({
+      type: 'list',
+      name: 'vm',
+      choices: ["None", "Docker", "Minikube"],
```
Comment (jarinykanen) [inline src/commands/new-proj.ts]:
```suggestion
      type: "list",
      name: "vm",
      choices: [ "None", "Docker", "Minikube" ],
```

File: src/commands/new-proj.ts [hunk 32]
```diff
@@ -1,44 +1,114 @@
...
+    if (vmResult.vm) {
+      projVm = vmResult.vm;
+
+    } else {
+      throw new Error("No environment option was given for the project");
+    }
+
+    const pathResult = await this.prompt({
+      type: 'input',
+      name: 'path',
```
Comment (jarinykanen) [inline src/commands/new-proj.ts]:
```suggestion
      type: "input",
      name: "path",
```

File: src/commands/new-proj.ts [hunk 33]
```diff
@@ -1,44 +1,114 @@
...
+  repoViaVorpal();
+
   try {
     const testResult = await this.prompt({
-      type : "confirm",
-      name : "testAnswer",
-      message : `do you want to run a test for ${projName}`
+      type: "confirm",
+      name: "testAnswer",
+      message: `do you want to run a test for ${projName}`
```
Comment (jarinykanen) [inline src/commands/new-proj.ts]:
```suggestion
      message: `Do you want to run a test for ${projName}`
```

File: src/commands/new-proj.ts [hunk 34]
```diff
@@ -47,7 +117,50 @@ async function action() {
...
+
+/**
+ * Inits a default project with no framework attached
+ */
+async function initDefaultProject() {
+  try {
+    const cmds: string[] = await CreateDefault(projName, folderPath, repoPath);
+    await runExecSync(cmds[0]);
+    await runExecSync(cmds[1], {cwd: `.${path.sep}resources`});
+    await runExecSync(cmds[2], {cwd: `.${path.sep}resources`});
```
Comment (jarinykanen) [inline src/commands/new-proj.ts]:
```suggestion
    await runExecSync(cmds[1], { cwd: `.${path.sep}resources` });
    await runExecSync(cmds[2], { cwd: `.${path.sep}resources` });
```

File: src/commands/new-repo.ts [hunk 35]
```diff
@@ -16,27 +21,22 @@ const defaultPath = `${HOME}/.meta-proj-cli/projects`;
...
+  
   if (!publicity || !repoName) {
     try {
       if (!repoName) {
         const nameResult = await this.prompt({
-          type : 'input',
-          name : 'name',
-          message : "Give a name for the repository (leave empty to cancel): "
+          type: 'input',
+          name: 'name',
```
Comment (jarinykanen) [inline src/commands/new-repo.ts]:
```suggestion
          type: "input",
          name: "name",
```

File: src/commands/new-repo.ts [hunk 36]
```diff
@@ -47,88 +47,80 @@ async function action(args) {
       }
 
       const publicityResult = await this.prompt({
-        type : 'list',
-        name : 'publicity',
-        choices : ["private", "internal", "public"],
-        message : "Set the publicity of the repository: "
+        type: 'list',
+        name: 'publicity',
+        choices: ["private", "internal", "public"],
```
Comment (jarinykanen) [inline src/commands/new-repo.ts]:
```suggestion
        type: "list",
        name: "publicity",
        choices: [ "private", "internal", "public" ],
```

File: src/commands/new-repo.ts [hunk 37]
```diff
@@ -47,88 +47,80 @@ async function action(args) {
...
-        name : 'description',
-        message : "Give a description for the repository: "
-      });
-  
-      const pathResult = await this.prompt({
-        type : 'input',
-        name : 'path',
-        message : "Set a path where to initiate repository, leave empty for default: "
+        type: 'input',
+        name: 'description',
```
Comment (jarinykanen) [inline src/commands/new-repo.ts]:
```suggestion
        type: "input",
        name: "description",
```

File: src/commands/new-repo.ts [hunk 38]
```diff
@@ -47,88 +47,80 @@ async function action(args) {
...
+async function finishRepo() {
+  try {
+    await runExecSync(
+      `gh repo create\
+      ${process.env.GIT_ORGANIZATION}/${repoName}\
+      --${publicity}\
+      ${description ? `-d="${description}"` : ""}\
+      ${template ? `--template="${process.env.GIT_ORGANIZATION}/${template}"` : ""}\
+      -y`,
+      {cwd: folderPath}
```
Comment (jarinykanen) [inline src/commands/new-repo.ts]:
```suggestion
      { cwd: folderPath }
```

File: src/commands/new-repo.ts [hunk 39]
```diff
@@ -47,88 +47,80 @@ async function action(args) {
...
+      await runExecSync(`git pull -q git@github.com:${process.env.GIT_ORGANIZATION}/${template}.git`, {cwd : repoPath});
+      await runExecSync("git branch -m master develop", {cwd: repoPath});
+      await checkout();
+    } else {
+      await runExecSync("git init", {cwd: repoPath});
+      await runExecSync("git checkout -q -b develop", {cwd: repoPath});
+      await runExecSync(`git add README.md`, {cwd: repoPath});
+      await runExecSync(`git commit -q -m "first commit"`, {cwd: repoPath});
+      await runExecSync(`git push -q origin develop`, {cwd: repoPath});
+      await checkout();
```
Comment (jarinykanen) [inline src/commands/new-repo.ts]: Check previous comments

File: src/commands/new-repo.ts [hunk 40]
```diff
@@ -47,88 +47,80 @@ async function action(args) {
...
 }
 
-async function finishRepo (repoPath) {
-  await runExecSync(`git checkout -q -b master`, {cwd : repoPath}); 
-  await runExecSync(`git push -q origin master`, {cwd : repoPath, stdio : ["ignore", "ignore", "ignore"]});
-  await runExecSync(`git checkout -q develop`, {cwd : repoPath});
+async function checkout() {
+  await runExecSync(`git checkout -q -b master`, {cwd: repoPath}); 
+  await runExecSync(`git push -q origin master`, {cwd: repoPath, stdio: ["ignore", "ignore", "ignore"]});
+  await runExecSync(`git checkout -q develop`, {cwd: repoPath});
```
Comment (jarinykanen) [inline src/commands/new-repo.ts]: Check previous comments

File: src/commands/pull-proj.ts [hunk 41]
```diff
@@ -12,17 +12,17 @@ const defaultPath = `${HOME}/.meta-proj-cli/projects`;
...
+  const copy: string = await OsUtils.getCommand("copy");
 
   try { 
     const repoNameResult = await this.prompt({
-      type : 'input',
-      name : 'name',
-      message : "Give the name of the repository: "
+      type: 'input',
+      name: 'name',
+      message: "Give the name of the repository: "
```
Comment (jarinykanen) [inline src/commands/pull-proj.ts]: Check previous comments

File: src/commands/pull-proj.ts [hunk 42]
```diff
@@ -35,16 +35,16 @@ async function action() {
   }
 
   try {
-    await runExecSync(`gh repo view ${process.env.GIT_ORGANIZATION}/${repoName}`, {stdio : "ignore"});
+    await runExecSync(`gh repo view ${process.env.GIT_ORGANIZATION}/${repoName}`, {stdio: "ignore"});
```
Comment (jarinykanen) [inline src/commands/pull-proj.ts]: Check previous comments

File: src/commands/pull-proj.ts [hunk 43]
```diff
@@ -35,16 +35,16 @@ async function action() {
...
     throw new Error(`Error while searching for repository: ${err}`);
   }
 
   try { 
     const repoIsLocalResult = await this.prompt({
-      type : 'confirm',
-      name : 'isLocal',
-      message : "Is the repository installed locally? "
+      type: 'confirm',
+      name: 'isLocal',
```
Comment (jarinykanen) [inline src/commands/pull-proj.ts]: Check previous comments

File: src/commands/pull-proj.ts [hunk 44]
```diff
@@ -55,9 +55,9 @@ async function action() {
 
   try { 
     const repoPathResult = await this.prompt({
-      type : 'input',
-      name : 'path',
-      message : repoIsLocal ?
+      type: 'input',
+      name: 'path',
```
Comment (jarinykanen) [inline src/commands/pull-proj.ts]: Check previous comments

File: src/commands/pull-proj.ts [hunk 45]
```diff
@@ -82,29 +82,29 @@ async function action() {
...
+    await runExecSync("git init", {cwd: repoPath});
 
-    await runExecSync(`${copy} project-config.json ${folderPath}`, {cwd : `.${path.sep}resources`});
+    await runExecSync(`${copy} project-config.json ${folderPath}`, {cwd: `.${path.sep}resources`});
 
     await runExecSync(
       `git remote add origin git@github.com:${process.env.GIT_ORGANIZATION}/${repoName}.git`,
-      {cwd : repoPath}
+      {cwd: repoPath}
     );
```
Comment (jarinykanen) [inline src/commands/pull-proj.ts]: Check previous comments

File: src/commands/select-os.ts [hunk 46]
```diff
@@ -7,10 +7,10 @@ import OsUtils from "../classes/os-utils";
 async function action() {
   try {
     const osResult = await this.prompt({
-      type : "list",
-      name : "os",
-      choices : ["LINUX", "WINDOWS", "MAC OS"],
-      message : "Which operating system are you using? "
+      type: "list",
+      name: "os",
+      choices: ["LINUX", "WINDOWS", "MAC OS"],
```
Comment (jarinykanen) [inline src/commands/select-os.ts]:
```suggestion
      choices: [ "LINUX", "WINDOWS", "MAC OS" ],
```

File: test/checkTest.spec.ts [hunk 47]
```diff
@@ -15,7 +15,7 @@ it('exports itself', () => {
 });
 
 it('checks for prequisites', async () => {
-  const test : CheckSet[] = [{checkable : "git", details : {}}]
+  const test: CheckSet[] = [{checkable: "git", details: {}}]
```
Comment (jarinykanen) [inline test/checkTest.spec.ts]:
```suggestion
  const test: CheckSet[] = [{ checkable: "git", details: {} }]
```

File: test/new-repo.spec.ts [hunk 48]
```diff
@@ -57,8 +56,8 @@ it('inits a Git project', () => {
   dotenv.config();
   const { HOME } = process.env;
   const defaultPath = `${HOME}/.meta-proj-cli/projects`;
-  child_process.execSync("git init", {cwd : (defaultPath + path.sep + "test")});
+  child_process.execSync("git init", {cwd: (defaultPath + path.sep + "test")});
```
Comment (jarinykanen) [inline test/new-repo.spec.ts]: Check previous comments

**Review by jarinykanen — CHANGES_REQUESTED** [review]

File: src/classes/path-utils.ts (hunk 25 above)
Comment (veikkahonkanen) [reply src/classes/path-utils.ts]: Moved tsdocs, added tsdocs

File: src/classes/path-utils.ts [hunk 49]
```diff
@@ -10,37 +10,58 @@ const defaultProjectPath = "~/.meta-proj-cli/projects";
...
-  public static savePath = async () : Promise<string> => {
-    const os : string = await PathUtils.osResolver();
+  public static savePath = async (): Promise<string> => {
+    const os: string = await PathUtils.osResolver();
     return await PathUtils.translatePath(defaultSavePath, os);
   }
 
-  public static projectPath = async () : Promise<string> => {
-    const os : string = await PathUtils.osResolver();
+  public static projectPath = async (): Promise<string> => {
```
Comment (jarinykanen) [inline src/classes/path-utils.ts]: Add docs

**Review by jarinykanen — CHANGES_REQUESTED** [review]

File: src/classes/path-utils.ts (hunk 49 above)
Comment (veikkahonkanen) [reply src/classes/path-utils.ts]: Added missing docs

**Review by jarinykanen — APPROVED** [review]

File: src/commands/add-readme.ts [hunk 50]
```diff
@@ -19,17 +19,17 @@ async function action(args: { options: { path: string; }; }) {
...
 
-  await runExecSync("git init", {cwd : path});
-  await runExecSync("git add README.md", {cwd : path});
-  await runExecSync('git commit -m "first commit"', {cwd : path});
-  await runExecSync(`git remote add origin ${process.env.GIT_REPO_BASE_PATH}`, {cwd : path});
-  await runExecSync("git push -u origin master", {cwd : path});
+  await runExecSync("git init", { cwd: path });
+  await runExecSync("git add README.md", { cwd: path });
+  await runExecSync('git commit -m "first commit"', { cwd: path });
+  await runExecSync(`git remote add origin ${process.env.GIT_REPO_BASE_PATH}`, { cwd: path} );
```
Comment (eetupur) [inline src/commands/add-readme.ts:28]:
```suggestion
  await runExecSync(`git remote add origin ${process.env.GIT_REPO_BASE_PATH}`, { cwd: path });
```

**Review by eetupur — CHANGES_REQUESTED** [review]
found one inconsistency

**Pull request merged**